### PR TITLE
support Mackerel Webhook

### PR DESCRIPTION
**Pull request by y-yagi**
Hi, I added support for mackerel webhook.
http://help-ja.mackerel.io/entry/howto/alerts

idobata_hook_mackerel

### CODE REVIEW

File: lib/hooks/mackerel/templates/default.html.haml [hunk 1]
```diff
@@ -0,0 +1,5 @@
+%p
+  %span.label{class: label_class_from_build_status(payload.alert.status)}= payload.alert.status.upcase
+  = "#{payload.alert.monitorName} at #{payload.host.name} (#{payload.host.status}) #{payload.host.roles.first.fullname}"
```
Comment (ursm) [inline lib/hooks/mackerel/templates/default.html.haml]:
``` haml
= "#{foo} bar"
```
is equivalent to:
``` haml
#{foo} bar
```

File: lib/hooks/mackerel/templates/default.html.haml [hunk 2]
```diff
@@ -0,0 +1,5 @@
...
+  %br
+  %a{href: payload.alert.url} #{payload.alert.url}
```
Comment (ursm) [inline lib/hooks/mackerel/templates/default.html.haml]:
``` haml
#{foo}
```
is equivalent to:
``` haml
= foo
```

**Comment by ursm** [comment]
Thanks! We'll test it in next week. Please wait a minute.

**Comment by y-yagi** [comment]
Thanks for reviewing!  I fixed!

**Comment by ursm** [comment]
How about indicating `(show)`, `(detail)` or something instead of using URL directly as a link text?
(like `(build)` and `(changeset)` in below)
selection_205

**Comment by y-yagi** [comment]
Thanks you for suggestion!  I think that `detail` is good. I updated.

idobata_hook_mackerel

**Comment by ursm** [comment]
Looks great. Your contribution is much appreciated.

File: lib/hooks/mackerel/templates/default.html.haml [hunk 3]
```diff
@@ -0,0 +1,5 @@
+%p
+  %span.label{class: label_class_from_build_status(payload.alert.status)}= payload.alert.status.upcase
+  #{payload.alert.monitorName} at #{payload.host.name} (#{payload.host.status}) #{payload.host.roles.first.fullname}
```
Comment (hibariya) [inline lib/hooks/mackerel/templates/default.html.haml]: AFAIK, `payload.host.roles` can be empty array. Would you please consider this? Like `payload.host.roles.first.try(:full_name)` and so on.

Comment (y-yagi) [inline lib/hooks/mackerel/templates/default.html.haml]: @hibariya Thanks for reviewing!  I overlook it, and I am sorry. I updated it in `payload.host.roles.first.try(:fullname)`.

**Comment by hibariya** [comment]
Now Mackerel hook is working well on our test server :fish: :sparkles: :sparkles: :sparkles: 

By the way, we would love to handle `sample` event of Webhook.
@y-yagi Can you add it's support..?

Mackerel sends following payload when test triggered.

```json
{"event":"sample","message":"Sample Notification from Mackerel"}
```

([This article](https://github.com/idobata/idobata-hooks#%E3%83%86%E3%83%B3%E3%83%97%E3%83%AC%E3%83%BC%E3%83%88%E3%82%92%E4%BD%BF%E3%81%84%E5%88%86%E3%81%91%E3%82%8B) and [this implementation](https://github.com/idobata/idobata-hooks/blob/master/lib/hooks/github/hook.rb#L22) should be useful for us.)

File: lib/hooks/mackerel/templates/default.html.haml (hunk 3 above)
Comment (hibariya) [inline lib/hooks/mackerel/templates/default.html.haml]: Thanks :whale:

**Comment by y-yagi** [comment]
@hibariya Thank you for test! 
I added support of `sample` event. Can you confirm it?

**Comment by hibariya** [comment]
We received below message without any problems. Yay :whale:

selection_167

**Comment by hibariya** [comment]
It seems like this is ready to deploy, so I'm going to merge this and apply to production server  :rocket:

**Comment by y-yagi** [comment]
Thank you very much for everything!!!!

**Comment by hibariya** [comment]
Now Mackerel hook is available on Idobata :star2: 

@y-yagi Thank you for your nice PR, too :fish: :dash:

**Pull request merged**